test/mlir/onnx/onnx_constprop_no_shape_inference.mlir

```diff
@@ -64,3 +64,36 @@ func @test_splitv11_axis_1_no_splitattr() -> (tensor<2x5xf32>, tensor<2x5xf32>)
   // CHECK: {{.*}}  = "onnx.Constant"() {value = dense<{{\[}}[5.000000e+00, 6.000000e+00, 7.000000e+00, 8.000000e+00, 9.000000e+00], [1.500000e+01, 1.600000e+01, 1.700000e+01, 1.800000e+01, 1.900000e+01]]> : tensor<2x5xf32>} : () -> tensor<2x5xf32>
   // CHECK-NOT: {{.*}} = "onnx.SplitV11"{{.*}}
 }
+
+//===----------------------------------------------------------------------===//
+/// scatternd tests
+
+// CHECK-LABEL: @test_scatternd_f32()
+func @test_scatternd_f32() -> (tensor<8xf32>) {
+  %0 = "onnx.Constant"() { name = "constant.0", value = dense<[1., 2., 3., 4., 5., 6., 7., 8.]>:tensor<8xf32> } : () -> tensor<8xf32>
+  %1 = "onnx.Constant"() { name = "constant.1", value = dense< [[4], [3], [1], [7]]>:tensor<4x1xi64> } : () -> tensor<4x1xi64>
+  %2 = "onnx.Constant"() { name = "constant.2", value = dense<[9., 10., 11., 12.]>:tensor<4xf32> } : () -> tensor<4xf32>
+  // CHECK : [[R1:%.+]] = "onnx.Constant"{{.*}} dense<{{\[}}1.000000e+00, 1.100000e+01, 3.000000e+00, 1.000000e+01, 9.000000e+00, 6.000000e+00, 7.000000e+00, 1.200000e+01],
+  // CHECK-NEXT : return [[R1]] : tensor<8xf32>
+  %3 = "onnx.ScatterND"(%0, %1, %2) {node_name = "ScatterND_6467", node_type = "ScatterND"} : (tensor<8xf32>, tensor<4x1xi64>, tensor<4xf32>) -> tensor<8xf32>
+  return %3 : tensor<8xf32>
+}
+
+// ----
```

Comment:
You need 5 `-`(s) instead of 4 to make `--split-input-file` work.

Reply:
Thanks for your kindly remind